ledger/complete/ledger.go

```diff
@@ -405,15 +407,7 @@ func (l *Ledger) ExportCheckpointAt(
 			payloadSize = newPayloadSize
 		}
 
-		l.logger.Info().Msgf("creating paths for %v payloads", len(payloads))
-
-		// get paths
-		paths, err := pathfinder.PathsFromPayloads(payloads, targetPathFinderVersion)
```

Comment:
Iterating through the paths again takes extra time. 

Since the `migrate` is going to return the payloads, we could let it return the path at the same time, so that computing the path could also be done concurrently.

Reply:
this seems wasteful.  you're recomputing the paths on migration iteration, and throwing away everything except the last iteration.  just parallelize the path computing here instead.